src/test/kotlin/uk/gov/justice/digital/hmpps/manageusersapi/resource/RolesControllerIntTest.kt

```diff
@@ -51,6 +58,138 @@ class RolesControllerIntTest : IntegrationTestBase() {
             "adminTypeName":"External Administrator"}]
           }
           """
-      )
+        )
+    }
+  }
+
+  @Nested
+  inner class AmendRoleName {
+
+    @Test
+    fun `Change role name endpoint not accessible without valid token`() {
+      webTestClient.put().uri("/roles/ANY_ROLE")
+        .exchange()
+        .expectStatus().isUnauthorized
+    }
+
+    @Test
+    fun `Change role name endpoint returns forbidden when does not have admin role `() {
+      webTestClient
+        .put().uri("/roles/ANY_ROLE")
+        .headers(setAuthorisation(roles = listOf()))
+        .body(BodyInserters.fromValue(mapOf("roleName" to "new role name")))
+        .exchange()
+        .expectStatus().isForbidden
+        .expectBody()
+        .jsonPath("$").value<Map<String, Any>> {
+          mapOf(
+            "status" to "403"
+          )
+        }
+    }
+
+    @Test
+    fun `Change role name returns error when role not found`() {
+      hmppsAuth.stubPutRoleNameFail("Not_A_Role", 404)
+      webTestClient
+        .put().uri("/roles/Not_A_Role")
+        .headers(setAuthorisation(roles = listOf("ROLE_ROLES_ADMIN")))
+        .body(BodyInserters.fromValue(mapOf("roleName" to "new role name")))
+        .exchange()
+        .expectStatus().isNotFound
+        .expectHeader().contentType(MediaType.APPLICATION_JSON)
+        .expectBody()
+        .jsonPath("status").isEqualTo("404")
+        .jsonPath("$").value<Map<String, Any>> {
+          assertThat(it["userMessage"] as String).isEqualTo("Unexpected error: Unable to get role: Not_A_Role with reason: notfound")
+          assertThat(it["developerMessage"] as String).isEqualTo("Unable to get role: Not_A_Role with reason: notfound")
+        }
+    }
+
+    @Test
+    fun `Change role name returns error when length too short`() {
+      webTestClient
+        .put().uri("/roles/OAUTH_ADMIN")
+        .headers(setAuthorisation(roles = listOf("ROLE_ROLES_ADMIN")))
+        .body(BodyInserters.fromValue(mapOf("roleName" to "tim")))
+        .exchange()
+        .expectStatus().isBadRequest
+        .expectBody()
+        .jsonPath("$").value<Map<String, Any>> {
+          assertThat(it["userMessage"] as String).isEqualTo("Validation failure: roleNameAmendment")
+          assertThat(it["developerMessage"] as String).contains("default message [roleName],100,4]")
+        }
```

Comment:
this is what I was mentioning earlier compare to court-register where you end u getting a list of nice messages back